client/src/main/java/org/eclipse/hono/client/CommandConnection.java

```diff
@@ -27,22 +27,32 @@
 public interface CommandConnection extends HonoClient {
 
     /**
-     * Gets a command consumer for a device.
+     * Creates a command consumer for a device.
      * <p>
-     * Implementations may choose to cache consumers for devices and
-     * return a cached instance instead of creating a new consumer on
-     * each invocation.
+     * For each device only one command consumer may be active at any given time.
+     * The {@link #closeCommandConsumer(String, String)} method can be used to close
+     * an active consumer.
```

Comment:
I think the wording should be a bit stronger, more in the direction "it is the responsibility of the caller to close the instance…"